packages/lib/slots.ts

```diff
@@ -220,24 +219,45 @@ function buildSlotsWithDateRanges({
         time: slotStartTime,
       };
 
-      if (dateOutOfOfficeExists) {
-        const { toUser, fromUser, reason, emoji } = dateOutOfOfficeExists;
-
-        slotData = {
-          time: slotStartTime,
-          away: true,
-          ...(fromUser && { fromUser }),
-          ...(toUser && { toUser }),
-          ...(reason && { reason }),
-          ...(emoji && { emoji }),
-        };
-      }
-
       slots.push(slotData);
       slotStartTime = slotStartTime.add(frequency + (offsetStart ?? 0), "minutes");
     }
   });
 
+  if (datesOutOfOffice) {
+    Object.keys(datesOutOfOffice).forEach(async (date) => {
+      const dateOutOfOffice = datesOutOfOffice[date];
+      const slotTime = dayjs(date).tz(timeZone).startOf("day");
+
+      const { toUser, fromUser, reason, emoji } = dateOutOfOffice;
+
+      const enrichedToUser = toUser ? await UserRepository.enrichUserWithItsProfile({ user: toUser }) : null;
```

Comment:
If toUser belongs to an org, redirecting to that user leads to a 404 page. This fixes that.

Bug Video: 

https://github.com/user-attachments/assets/73212f8b-71cb-4431-8e49-7e45c9b406d7